Scripts/prepare-coverage-reports.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/zsh -l
+set -e
+
+function exportlcov() {
+    build_type=$1
+    executable_name=$2
+
+    executable=$(find "${directory}" -type f -name $executable_name)
+    profile=$(find "${directory}" -type f -name 'Coverage.profdata')
+    output_file_name="$executable_name.lcov"
+
+    can_proceed=true
+    if [[ $build_type == watchOS* ]]; then
+        echo "\tAborting creation of $output_file_name – watchOS not supported."
+    elif [[ -z $profile ]]; then
+        echo "\tAborting creation of $output_file_name – no profile found."
+    elif [[ -z $executable ]]; then
+        echo "\tAborting creation of $output_file_name – no executable found."
```

Comment:
Should we fail in these cases?

Reply:
Not necessarily. Since I put `watchOS` at the top we should never hit these cases. But it seems reasonable that we'd run jobs that wouldn't produce code coverage data – we're not being smart about what build types we apply this prepare step on.